### ISSUE TITLE
SSL on IPv6

### ISSUE THREAD

**Issue by starburst997** [HaxeFoundation/haxe#6222]
Following #3122, SSL seems to be getting mandatory for iOS ?

https://nabla-c0d3.github.io/blog/2016/08/14/ats-enforced-2017/

https://developer.apple.com/news/?id=12212016b

Currently, I am not able to get SSL working on IPv6. It just failed with an EOF error.

Also SSL on iOS using IPv4, trigger an error ([issue](https://github.com/HaxeFoundation/hxcpp/issues/570)), "No CA Chain is set, but required to operate", at the very least this one seems to be fixable by providing our own cert (not ideal...).

I guess if Apple decide to enforce it tomorrow, it's always possible to quickly hack an extension or something, but this seem like something haxe should be able to do natively.

**Comment by starburst997** [comment]
Copy / Paste from #3122 

I created a repo to do some tests on IPV6 and SSL.

https://github.com/starburst997/haxe-openfl-ipv6-ssl

I test Http, [Akifox](https://github.com/yupswing/akifox-asynchttp) (because it uses Socket) and URLLoader

I'm using OpenFL 4 (didn't try other version)

To test IPV6 I used this guide: [Mac Internet Sharing](https://developer.apple.com/library/content/documentation/NetworkingInternetWeb/Conceptual/NetworkingOverview/UnderstandingandPreparingfortheIPv6Transition/UnderstandingandPreparingfortheIPv6Transition.html#//apple_ref/doc/uid/TP40010220-CH213-SW16)

***My results are as follow (with applying PR #6142):***

- [IPV6](https://github.com/starburst997/haxe-openfl-ipv6-ssl/tree/master/results/IPV6_PR6142.md)
- [IPV4](https://github.com/starburst997/haxe-openfl-ipv6-ssl/tree/master/results/IPV4_PR6142.md)

*Basically*:

HTTPS on IPV6 doesn't works, ***EOF***

***On IOS***: OpenFL's URLLoader seems to be able to do the job just fine, IPV6 + SSL, and it's the only one working (but failed with Error: 6 on other target)
